src/components/EventList/components/EventItem/index.jsx

```diff
@@ -1,10 +1,59 @@
-import { CardContent, Collapse, Grid, Typography } from '@mui/material';
+import { Button, CardContent, Collapse, Grid, Typography } from '@mui/material';
 import { useState } from 'react';
 import styles from './EventItem.module.css';
+import { AddToCalendarButton } from 'add-to-calendar-button-react';
 
 const EventItem = ({ event }) => {
   const [expanded, setExpanded] = useState(false);
 
+  const convertDate = (dateString) => {
+    const dateParts = dateString.split(' ');
+    const month = dateParts[0];
+    const day = dateParts[1].slice(0, -1); // Remove the comma at the end
+    const year = dateParts[2];
+  
+    // Convert month abbreviation to month number
+    const monthIndex = new Date(Date.parse(`${month} 1, ${year}`)).getMonth() + 1;
+    const monthString = monthIndex.toString().padStart(2, '0');
+    const dayString = day.padStart(2, '0');
+  
+    return `${year}-${monthString}-${dayString}`;
+  };
+  const convertTime = (timeString) => {
+    const times = timeString.split(' - ');
+    const startTime = times[0];
+    const endTime = times[1];
+  
+    const convertSingleTime = (singleTime) => {
+      const [time, period] = singleTime.split(' ');
+      const [hours, minutes] = time.split(':');
+      let convertedHours = parseInt(hours, 10);
+      
+      if (period.toLowerCase() === 'pm' && convertedHours !== 12) {
```

Comment:
This seems like a lot of work just to get datetime objects in a specific format. There are libraries that will do this for you instead of you reinventing the wheel a bit. We have a package installed in 360 called `moment`.

See:
https://momentjs.com/docs/#/parsing/
https://momentjs.com/docs/#/displaying/format/

Additionally, why do you need to format the date/time at all. I would be very surprised if `AddToCalendarButton` can not take in a date object.

Reply:
The startDate and endDate only accept format YYYY-MM-DD, so I have to call convertDate in AddToCalendarButton

Reply:
Thanks for the useful links! I would notice that next time

Reply:
> The startDate and endDate only accept format YYYY-MM-DD, so I have to call convertDate in AddToCalendarButton

My proposal is that you don't reinvent the wheel by creating a `convertDate` function. That work is already done for you. Please see the links that I sent.

Reply:
> Thanks for the useful links! I would notice that next time

This was a comment on your current PR while I am requesting be addressed before merge. This should be considered now, not next time. I welcome @amos-cha 's feedback as well.

Reply:
> This seems like a lot of work just to get datetime objects in a specific format. There are libraries that will do this for you instead of you reinventing the wheel a bit. We have a package installed in 360 called `moment`.
> 
> 
> 
> See:
> 
> https://momentjs.com/docs/#/parsing/
> 
> https://momentjs.com/docs/#/displaying/format/
> 
> 
> 
> Additionally, why do you need to format the date/time at all. I would be very surprised if `AddToCalendarButton` can not take in a date object.

Moment is deprecated. We use date-fns instead, which has an equally powerful format function

Reply:
Sure, the date-fns docs for the `format` function are here: https://date-fns.org/v2.30.0/docs/format

Reply:
@EjPlatzer Is the date-fns already in the 360 package as moment? Or should I install it instead

Reply:
You can confirm that it is installed in `package.json`.
![image](https://github.com/gordon-cs/gordon-360-ui/assets/43764277/cf145431-904c-42b3-ba61-9756f6cdb57f)